### PR TITLE
Upgrade k8s to v1.17.1

### PR DESCRIPTION
**Pull request by nickolaev**
Upgrade to the latest Kubernetes release.
Remove the stale sriov controller code while passing.

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: k8s/go.mod [hunk 1]
```diff
@@ -32,30 +33,30 @@ replace (
...
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191114111229-2e90afcb56c7
-	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191114113550-6123e1c827f7
-	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191114110954-d67a8e7e2200
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20191114112655-db9be3e678bb
-	k8s.io/metrics => k8s.io/metrics v0.0.0-20191114105837-a4a2842dc51b
-	k8s.io/node-api => k8s.io/node-api v0.0.0-20191114112948-fde05759caf8
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191114104439-68caf20693ac
-	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.0.0-20191114110435-31b16e91580f
-	k8s.io/sample-controller => k8s.io/sample-controller v0.0.0-20191114104921-b2770fad52e3
+	k8s.io/api => k8s.io/api v0.17.0
```
Comment (edwarnicke) [inline k8s/go.mod]: Do we still need all these replaces?

File: k8s/pkg/apis/go.mod [hunk 2]
```diff
@@ -2,6 +2,31 @@ module github.com/networkservicemesh/networkservicemesh/k8s/pkg/apis
 
 go 1.13
 
-require k8s.io/apimachinery v0.0.0
+require k8s.io/apimachinery v0.17.0
 
-replace k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20191028221656-72ed19daf4bb
+replace (
+	k8s.io/api => k8s.io/api v0.17.0
```
Comment (edwarnicke) [inline k8s/pkg/apis/go.mod]: Do we still need all these replaces?

File: test/go.mod [hunk 3]
```diff
@@ -40,30 +40,30 @@ replace (
...
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191114111229-2e90afcb56c7
-	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191114113550-6123e1c827f7
-	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191114110954-d67a8e7e2200
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20191114112655-db9be3e678bb
-	k8s.io/metrics => k8s.io/metrics v0.0.0-20191114105837-a4a2842dc51b
-	k8s.io/node-api => k8s.io/node-api v0.0.0-20191114112948-fde05759caf8
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191114104439-68caf20693ac
-	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.0.0-20191114110435-31b16e91580f
-	k8s.io/sample-controller => k8s.io/sample-controller v0.0.0-20191114104921-b2770fad52e3
+	k8s.io/api => k8s.io/api v0.17.0
```
Comment (edwarnicke) [inline test/go.mod]: Do we still need these replaces?

File: k8s/pkg/apis/go.mod (hunk 2 above)
Comment (nickolaev) [reply k8s/pkg/apis/go.mod]: I tried to remove these but it was not working immediately. If I get it right they are on the way to "fix" the whole versioning story but are not quite there yet.

File: test/go.mod [hunk 4]
```diff
@@ -41,30 +36,30 @@ replace (
...
-	k8s.io/node-api => k8s.io/node-api v0.0.0-20191114112948-fde05759caf8
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191114104439-68caf20693ac
-	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.0.0-20191114110435-31b16e91580f
-	k8s.io/sample-controller => k8s.io/sample-controller v0.0.0-20191114104921-b2770fad52e3
+	k8s.io/api => k8s.io/api v0.17.1
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.1
+	k8s.io/apimachinery => k8s.io/apimachinery v0.17.1
+	k8s.io/apiserver => k8s.io/apiserver v0.17.1
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.17.1
+	k8s.io/client-go => k8s.io/client-go v0.17.1
```
Comment (artem-belov) [inline test/go.mod]: Does k8s still require replaces?

**Pull request merged**